rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -88,25 +122,20 @@ void SequentialWriter::open(
   }
 
   const auto storage_uri = format_storage_uri(base_folder_, 0);
-
   storage_ = storage_factory_->open_read_write(storage_uri, storage_options.storage_id);
   if (!storage_) {
     throw std::runtime_error("No storage could be initialized. Abort");
   }
 
-  if (max_bagfile_size_ != 0 &&
-    max_bagfile_size_ < storage_->get_minimum_split_file_size())
-  {
-    throw std::runtime_error(
-            "Invalid bag splitting size given. Please provide a different value.");
-  }
-
-  init_metadata();
+  check_bagfile_size(storage_options);
+  init_compression(compression_options);
```

Comment:
Just a note. Not sure what can be done at this point since it seems to be baked into the design of this class, but this is quite a bit of `void` functions that throw.

Reply:
My thoughts are that each of these are a different step when calling `open()` that make sure the different components are setup correctly and throw when they fail to otherwise.

Reply:
I mean the issue is more with `SequentialWriter::open()` being a `void` function, so the only way to propagate failure from your sub-functions is to throw at some point.